### ISSUE TITLE
ModelCheckpoint Callback Doesn't Log Validation Accuracy on ML Engine

### ISSUE THREAD

**Issue by bclayman** [tensorflow/tensorflow#23469]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ML Engine Instance (Debian)
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 1.10.0
- Python version: 2.7.14
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A

**Describe the current behavior**

I've set up a CNN and have created callbacks for model checkpointing, TensorBoard visualization, and early stopping.  I have ~408K samples in my `tf.data.Dataset`, am using a batch size of 256, and train my model with 1,000 steps per epoch with 1000 epochs.  When I do this, everything trains correctly on ML Engine but I don't get logging like I normally do.

**Describe the expected behavior**

I should see logging on ML Engine that indicates validation accuracy for each epoch, like this:

Epoch 2/2000
1/200 [..............................] - ETA: 2:00 - loss: 5.7345 - acc: 0.1055

2/200...

When I set `steps_per_epoch` to be only 200, I *do* see validation accuracy logging as expected.  For what it's worth, the training still works and produces .h5 files.  But it's crucial that I be able to view logging as the model's training.

I spoke with folks on the ML Engine team and it's not an issue on their end.  They suspect it's a bug with the `tf.keras.callbacks.ModelCheckpoint` callback or me misusing `steps_per_epoch` somehow.  Is there a certain value I shouldn't exceed with `steps_per_epoch`?  I wanted to do one full pass over the data, so that should be: # samples / (steps_per_epoch * batch_size), which would be ~1,593 steps.  Even 1,000 steps_per_epoch causes this issue though.

**Code to reproduce the issue**

Unfortunately, I can't share the data I'm using.  I'm using a CNN built with `tf.keras.layers` layers (called `model`) and my compile step, callbacks definitions, and fit step look like this:

        model.compile(
            loss='sparse_categorical_crossentropy',
            optimizer='adam',
            metrics=['accuracy'],
        )
        callbacks = [
            ModelCheckpoint("model_{epoch:04d}_{val_acc:.4f}.h5",
                            monitor='val_acc',
                            verbose=1,
                            save_best_only=True,
                            mode='max'),
            TensorBoard(),
            EarlyStopping(monitor='val_acc', patience=5, min_delta=0, mode='max')
        ]

        model.fit(dataset, steps_per_epoch=1000, epochs=1000, validation_data=val_dataset,
                  validation_steps=3, callbacks=callbacks)

**Other info / logs**

Here's what I see in the ML Engine logs:

`14:24 Epoch 1/4`

`15:00 Starting evaluation`

With no logging in between 👎

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rmothukuru** [comment]
@bclayman,
Sorry for the delayed response. Can you please confirm if your issue is resolved?

Also, the argument, `steps_per_epoch` is not Mandatory. So, you can try leaving it empty and the value will be derived as stated below, in the [Documentation of Model.fit (steps_per_epoch)](https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit):

> When training with input tensors such as TensorFlow data tensors, the default None is equal to the number of samples in your dataset divided by the batch size, or 1 if that cannot be determined. If x is a tf.data dataset, and 'steps_per_epoch' is None, the epoch will run until the input dataset is exhausted. When passing an infinitely repeating dataset, you must specify the steps_per_epoch argument. This argument is not supported with array inputs.
